src/RoadRegistry.BackOffice/Uploads/RoadSegmentSurfaceChangeDbaseRecordsValidator.cs

```diff
@@ -27,15 +28,20 @@ public ZipArchiveProblems Validate(ZipArchiveEntry entry, IDbaseRecordEnumerator
                         var record = records.Current;
                         if (record != null)
                         {
+                            RecordType recordType = default;
                             if (!record.RECORDTYPE.HasValue)
                             {
                                 problems += recordContext.RequiredFieldIsNull(record.RECORDTYPE.Field);
                             } else
                             {
-                                if (!RecordType.ByIdentifier.ContainsKey(record.RECORDTYPE.Value))
+                                if (!RecordType.ByIdentifier.TryGetValue(record.RECORDTYPE.Value, out recordType))
                                 {
                                     problems += recordContext.RecordTypeMismatch(record.RECORDTYPE.Value);
                                 }
+                                else if (!recordType.IsAnyOf(RecordType.Identical, RecordType.Added, RecordType.Removed))
```

Comment:
Protect against any accidental modify record types (which are not going to be handled).